### PR TITLE
Start migration to Mixin framework.

### PR DESCRIPTION
**Pull request by Zidane**
As promised, I've made sure to keep the edits to a minimum so that the noise in this PR is kept equally so. I've included two mixins in the PR only to give you the ability to run and see that it does indeed work. I have tested dev time and production time and have equally working results.

### CODE REVIEW

**Comment by Niko-sk2x** [comment]
I know the previous code didn't have that, but while editing build.gradle you could add equivalent of [this](https://github.com/OpenCubicChunks/CubicChunks/blob/72dff42b2f98b1c40436ddb577ddff4a8f21bc6b/build.gradle.kts#L133-L147) to make `genIntellijRuns` generate the right JVM args automatically.

**Comment by Zidane** [comment]
@Barteks2x 

My focus was only to get Mixin operational in both dev and prod (as stated in the OP). I don't want to stray away from that goal.

Maybe @Ordinastie would appreciate a general build.gradle cleanup? There is a lot of stuff that could be done..

**Comment by Ordinastie** [comment]
By all means, clean :p

File: build.gradle [hunk 1]
```diff
@@ -9,9 +9,15 @@ buildscript {
 			name = 'forge-repo'
 			url = 'http://files.minecraftforge.net/maven/'
 		}
+		maven {
+			name = 'sponge'
+			url = 'http://repo.spongepowered.org/maven'
```
Comment (kashike) [inline build.gradle:14]: `https`, unnecessary redirect from `http` to `https` otherwise

**Pull request merged**